### DISCUSSION TITLE
Raspberry Pi Pico support?

### DISCUSSION THREAD

**Discussion by hadess** [necroware/gameport-adapter#75] [Ideas]
Waveshare makes a Raspberry Pi Pico with the same form factor as the Arduino Pro Micro used for this project, the RP2040-Zero:
https://www.waveshare.com/rp2040-zero.htm

I found it locally for around 6€.

This could allow for more headroom in terms of speed and, from what I can find, a similar price point to the Arduino Pro Micro. Is that something that would be interesting for the project longer-term?

**Comment by necroware** [comment]
I did already make some experiments with the Pi Pico, which I didn't open yet. It is faster and better in many ways, but for this project it doesn't look to be really necessary so far. I mean we can port the project to Pi Pico, but I don't see the point to jump on another platforms all the time. I can't afford to maintain and test this project for all the joysticks and MCUs out there. As long as the Pro Micro is sufficient for this project, I'd say we don't need to change the MCU. Should we run into problems one day, we'd still be able to change it, but then there would be no way back to the Pro Micro.

**Reply by hadess** [reply]
Fair enough. The problem I ran into was pricing for the Arduino Pro Micro, which seemed quite a bit more expensive than Pico variants. I guess it's still cheaper than Teensies I have used in the past.
